src/systems.rs

```diff
@@ -337,18 +335,41 @@ pub(crate) fn particle_transform(
                 time.raw_delta_seconds()
             };
 
-            // Apply acceleration
-            velocity.0 += particle.acceleration * delta_time;
+            // inititalize precalculated values
+            let mut ppv = PrecalculatedParticleVariables::new();
 
-            // Apply drag
-            let current_drag = particle.drag.at_lifetime_pct(lifetime_pct);
-            if current_drag > 0.0 {
-                let drag_force = velocity.0.length_squared() * current_drag * delta_time;
-                let drag_force = -velocity.0.normalize() * drag_force;
-                velocity.0 += drag_force;
-            }
+            // Apply velocity modifiers to velocity
+            for modifier in &particle.velocity_modifiers {
+                use VelocityModifier::{Constant, Drag, Noise, ScalarAcceleration};
+                match modifier {
+                    Constant(v) => {
+                        velocity.0 += *v * delta_time;
+                    }
+
+                    ScalarAcceleration(v) => {
```

Comment:
It might be useful to also have a plain `Acceleration` variant that is a `Vec3`. I think you mentioned it elsewhere, was the main blocker from doing that that we don't have a lerp for `Vec3`?

Reply:
There is already a Vec3 variant that was poorly named Constant.
I will rename this Vector, and ScalarAcceleration -> Scalar.
There was no blocker, but indeed I mentioned that it would be cool to have key values for Vec3 and f32. Not only lerp, which is cool, but don't offer as much control as key values (like gradient for ColorOverTime).